anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/ModernOrder.java

```diff
@@ -0,0 +1,100 @@
+package com.iluwatar.corruption.system.modern;
+
+import java.util.Objects;
+
+/**
+ * The class represents an order in the modern system.
+ */
+public class ModernOrder {
+  private String id;
+  private Customer customer;
+
+  private Shipment shipment;
+
+  private String extra;
+
+  /**
+   * Constructor.
+   * @param id the id of the order
+   * @param customer the customer of the order
+   * @param shipment the shipment of the order
+   * @param extra  the extra data of the order
+   */
+  public ModernOrder(String id, Customer customer, Shipment shipment, String extra) {
```

Comment:
Use Lombok to get rid of boilerplate such as getters, setters, constructors, etc.